smarts/core/opendrive_road_network.py

```diff
@@ -0,0 +1,1810 @@
+            ):
+                road_elem_speed = float(road_elem.types[0].speed.max)
+            else:
+                road_elem_speed = self._default_lane_speed
+
+            # Create new road for each lane section
+            for section_elem in road_elem.lanes.lane_sections:
+                section_elem: LaneSectionElement = section_elem
+
+                # Create new roads so that all lanes for each road are in same direction
+                for sub_road, suffix in [
+                    (section_elem.leftLanes, "L"),
+                    (section_elem.rightLanes, "R"),
+                ]:
+                    # Skip if there are no lanes
+                    if not sub_road:
+                        continue
+
+                    road_id = OpenDriveRoadNetwork._elem_id(section_elem, suffix)
+                    total_lanes = len(sub_road)
+
+                    road = OpenDriveRoadNetwork.Road(
+                        road_id,
+                        section_elem.parentRoad.junction is not None,
+                        section_elem.length,
+                        section_elem.sPos,
+                        total_lanes,
+                    )
+
+                    self._roads[road_id] = road
+                    assert road_id not in self._surfaces
+                    self._surfaces[road_id] = road
+
+                    for lane_elem in sub_road:
+                        lane_id = OpenDriveRoadNetwork._elem_id(lane_elem, suffix)
+                        lane = OpenDriveRoadNetwork.Lane(
+                            self,
+                            lane_id,
+                            road,
+                            lane_elem.id,
+                            section_elem.length,
+                            lane_elem.type.lower()
+                            in [
+                                "driving",
+                                "exit",
+                                "entry",
+                                "offramp",
+                                "onramp",
+                                "connectingramp",
+                            ],
+                            road_elem_speed,
+                            road_elem.planView,
+                        )
+                        # Set road as drivable if it has at least one lane drivable
+                        if not road.is_drivable:
+                            road.is_drivable = (
+                                lane_elem.type.lower()
+                                in [
+                                    "driving",
+                                    "exit",
+                                    "entry",
+                                    "offramp",
+                                    "onramp",
+                                    "connectingramp",
+                                ],
+                            )
+
+                        self._lanes[lane_id] = lane
+                        self._all_lanes.append(lane)
```

Comment:
`self._all_lanes` to preserve a specific order of lanes when building the `RTree`. But i guess we can also use `self._lanes.values()` Since that order is also preserved.